sample/awssqs/flow.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: flows.knative.dev/v1alpha1
+kind: Flow
+metadata:
+  name: aws-sqs-example
+spec:
+  serviceAccountName: feed-sa
+  trigger:
+    eventType: aws-sqs
+    resource: us-east-1/knative-demo
+    service: aws-sqs
+    parametersFrom:
+      - secretKeyRef:
```

Comment:
Should consider the case that cluster is running on AWS, and use KIAM to do IAM role delegation. This is how people access AWS resources from a k8s cluster running on EC2 instances without using any credentials.

The implementation can support both cases, with credential, or KIAM delegated Role. For the latter case, the IAM role configuration would be like:

1. Kubernetes cluster master nodes IAM role (This should already be existing after the cluster is created), this role should have the policy to assume-role

2. An IAM role which is assigned to a namespace (and will be used by the POD), this role is set to trust cluster master node IAM role, with KIAM, your pod can get temp credential to this IAM role by calling AWS metadata API, without using any credential. This Role also needs the policy to assume-role.

3. A third role to be able to operate SQS, and it trusts the 2nd IAM role.

4. In flow configuration, the 2nd and 3rd IAM role ARNs are needed, so that in your receive_adaptor you can use the 2nd role credential to assume 3rd role, and then read messages.

Reply:
I didn't see this PR until this afternoon. Before that, I implemented a SQS event source about 1 week ago, which was very similar to this. The major difference is, my k8s cluster is running on EC2 instances, and I use KIAM, so I don't use any explicit credentials to access AWS SQS. So I gave above comments, it would be better if this case can be considered in the PR.